### DISCUSSION TITLE
immich-machine-learning container is always unhealthy

### DISCUSSION THREAD

**Discussion by vnwnv** [immich-app/immich#12572] [Q&A] [answered]
my Immich-machine-learning container always unhealthy, here is my docker-compose config

```docker-compose
  immich-server:
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    container_name: immich-server
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    depends_on:
      - immich-redis
    restart: always
    networks:
      - webbackend
  immich-machine-learning:
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    container_name: immich-machine-learning
    volumes:
      - model_cache:/cache
    env_file:
      - .env
    environment:
      - HTTP_PROXY=http://192.168.1.20:10801
      - HTTPS_PROXY=http://192.168.1.20:10801
      - http_proxy=http://192.168.1.20:10801
      - https_proxy=http://192.168.1.20:10801
    restart: always
    networks:
      - webbackend
  immich-postgres:
    container_name: immich-postgres
    image: tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - immich_pgdata:/var/lib/postgresql/data
    restart: always
    networks:
      - webbackend
  immich-redis:
    container_name: immich-redis
    image: redis:7-bookworm
    volumes:
      - immich_redisdata:/data
    restart: always
    networks:
      - webbackend
```
and my env file

```ini
# The location where your uploaded files are stored
UPLOAD_LOCATION=/media/storage_01/public_store/immich/upload

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secrets for postgres and typesense. You should change these to random passwords
# TYPESENSE_API_KEY=<mykey>
DB_PASSWORD=<mydbpassword>

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich-postgres
DB_USERNAME=immich
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich-redis
```

I'm using proxy to reach the huggingface. the environment variable works correctly, the container can download models form huggingface.

I can use immich web to search the photo, and use face recognition function.

the healthcheck.py is
```python
import os
import sys

import requests

port = os.getenv("IMMICH_PORT", 3003)

try:
    response = requests.get(f"http://localhost:{port}/ping", timeout=2)
    if response.status_code == 200:
        sys.exit(0)
    sys.exit(1)
except requests.RequestException:
    sys.exit(1)
```

I run the `requests.get(f"http://localhost:3003/ping", timeout=2)` manually, the status code is 503.

I have tried delete the container and model cache volume, than recreate container, but it dose not work.

here are the logs after recreate container

```
[09/11/24 13:05:22] INFO     Starting gunicorn 23.0.0                           
[09/11/24 13:05:22] INFO     Listening at: http://[::]:3003 (8)                 
[09/11/24 13:05:22] INFO     Using worker: app.config.CustomUvicornWorker       
[09/11/24 13:05:22] INFO     Booting worker with pid: 9                         
[09/11/24 13:05:26] INFO     Started server process [9]                         
[09/11/24 13:05:26] INFO     Waiting for application startup.                   
[09/11/24 13:05:26] INFO     Created in-memory cache with unloading after 300s  
                             of inactivity.                                     
[09/11/24 13:05:26] INFO     Initialized request thread pool with 4 threads.    
[09/11/24 13:05:26] INFO     Application startup complete.                      
```

**Comment by mmomjian** [comment, marked as answer]
I assume that the http request is being routed through the proxy and thus failing, if I had to guess.

**Reply by vnwnv** [reply]
Thank you, you are right!
After i removed these proxy env variable, it works! maybe i need something like no_proxy.

**Reply by ChishFoxcat** [reply]
Why did the immich-machine-learning container fail because of the proxy?

**Reply by JuliusLedoux** [reply]
I am having the same issue my machine learning shows unhealthy and the logs are identical to the ones you have.
This is my compose file:
  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}-cuda
    deploy:
     resources:
       reservations:
         devices:
           - driver: nvidia
             count: 1
             capabilities:
               - gpu   
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - stack.env
    restart: always
    healthcheck:
      disable: false


I also had ran it without the GPU option (vanilla deployment) and still shows unhealthy. Any clues? I am running the current version as March 31, 2025 for immich.

**Reply by alextran1502** [reply]
@JuliusLedoux there is a regression, we will likely push out a fix tomorrow. You can use tag `v1.130.3` on machine-learning container for the time being

**Reply by sileclercq** [reply]
same issue as JuliusLedoux

Here is my config
  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - config.env
    restart: always
    healthcheck:
      disable: false
